src/components/functional_module/include/functional_module/plugin_manager.h

```diff
@@ -48,15 +48,27 @@ class PluginManager : public ModuleObserver {
   typedef std::map<ModuleID, ModulePtr> Modules;
   int LoadPlugins(const std::string& plugin_path);
   void UnloadPlugins();
-  void ProcessMessage(application_manager::MessagePtr msg);
+
+  /**
+   * @brief ProcessMessage forwards modules message to modules if any is
+   * subsribed for the message.
+   * @param msg Mobile message to process
+   * @return Result of processing
+   */
+  ProcessResult ProcessMessage(application_manager::MessagePtr msg);
+
+  /**
+   * @brief ProcessHMIMessage forwards HMI message to modules if any is
+   * subscribed for the message
+   * @param msg HMI message to process
+   * @return Result of processing
+   */
   ProcessResult ProcessHMIMessage(application_manager::MessagePtr msg);
   void SetServiceHandler(application_manager::ServicePtr service) {
     service_ = service;
   }
-  bool IsMessageForPlugin(application_manager::MessagePtr msg);
```

Comment:
@dev-gh I am not completely agree with such approch. 
It is good to check if plugin is able to process message before actualy processing it.

Reply:
@dev-gh so, use IsMessageForPlugin inside ProcessMessage. It is good public interface , and I dont think we should remove it

Reply:
@LuxoftAKutsan 
Addressed with #256